lib/hq/graphql/ext/object_extensions.rb

```diff
@@ -96,7 +96,16 @@ def field_from_column(column, auto_nil:)
             name = column.name
             return if field_exists?(name)
 
-            field name, Types.type_from_column(column), null: !auto_nil || column.null
+            field name, Types.type_from_column(column), null: !auto_nil || column.null,
+            authorize: -> (_obj, ctx) do
+              restriction = ctx[:current_user]&.restrictions&.detect do |el|
+                (el.resource.name == name || el.resource.alias == name) &&
+                el.restriction_operation_id == "HasHelpers::RestrictionOperation::::View" &&
+                el.resource.resource_type_id != "HasHelpers::ResourceType::::RequiredField"
```

Comment:
Why do we need to check RequiredField to view?

Reply:
required fields can't be restricted because this will cause a schema conflict. For example if attribute name is required, the schema assures that this attribute will always be returned as string. If it has a restriction, it will try to return null, causing a query fail for schema error